### ISSUE TITLE
`check-licenses`: GPLv3 isn't recognized as a compatible license

### ISSUE THREAD

**Issue by cliffordp** [WordPress/gutenberg#20701]
Re: https://github.com/WordPress/gutenberg/blob/%40wordpress/scripts%407.1.2/packages/scripts/scripts/check-licenses.js#L36

May be related: #7821

**Describe the bug**

GPLv3+ should be recognized as compatible, per:
A)
https://github.com/WordPress/WordPress/blob/5.3.2/readme.html#L94-L95
> WordPress is free software, and is released under the terms of the <abbr>GPL</abbr> (GNU General Public License) version 2 or (at your option) any later version.

and

B)
https://www.gnu.org/licenses/license-list.en.html#GNUGPL
> Please note that GPLv3 is not compatible with GPLv2 by itself. However, most software released under GPLv2 allows you to use the terms of later versions of the GPL as well. When this is the case, you can use the code under GPLv3 to make the desired combination.
> ... links to https://www.gnu.org/licenses/gpl-faq.html#AllCompatibility
> ... screenshot of it:

![image](https://user-images.githubusercontent.com/1812179/76157683-c0f5ff80-60d1-11ea-8b83-621e3a9e4c0c.png)

**To reproduce**

1)
In your plugin's `package.json`, set `"license": "GPL-3.0-or-later",` then run `wp-scripts check-licenses` and you'll get this message:
>  ERROR Module _your-plugin_ has an incompatible license 'GPL-3.0-or-later'.

2)
Change the package.json to `"license": "GPL-2.0-or-later",` re-run, and no messages.

**Expected behavior**

No error message with GPLv3+.

**Comment by gziolo** [comment]
@ntwb or @pento – do we need a fix in `@wordress/scripts`?

**Comment by ntwb** [comment]
I don’t think so, there’s a prior discussion on this subject, I’ll try dig this out tomorrow and follow up

**Comment by cliffordp** [comment]
Checking in... would be really curious if this is determined not to be an issue. Thanks!

**Comment by kevin940726** [comment]
@ntwb @cliffordp Do we have an update on this? Do we still want to fix this?

**Comment by cliffordp** [comment]
awaiting @ntwb's research from April 20

**Comment by ntwb** [comment]
I'm still unsure of this, search WP Slack for GPLv3 and there are numerous discussions in the theme and plugin channels on the subject, need to extract out this conversation

Possibly @Otto42 could weigh in here also

**Comment by ntwb** [comment]
@pento Any thoughts on this?

**Comment by pento** [comment]
`check-licenses` is a somewhat WordPress-specific tool: it assumes that the software is being released as GPLv2. Historically, WordPress hasn't used libraries, as defined in GPLv3: we've copied entire libraries, but we often modify them, which means they're considered copies, rather than libraries.

Gutenberg is better about this: libraries it uses are actually included as libraries, so GPLv3 could be added to `otherOssLicenses`. As I understand @cliffordp's original report, this would likely solve his particular issue, assuming he isn't using the `--gpl2` option.

It may also be worth being more explicit about the `check-licenses` behaviour in the `@wordpress/scripts/` readme: in particular, noting that this is a WordPress-specific tool, and isn't guaranteed to produce correct results in other use cases: licenses are usually only added the list when Gutenberg needs it.

Given that the plugin directory allows plugins to have [any GPL-compatible license](https://developer.wordpress.org/plugins/wordpress-org/detailed-plugin-guidelines/#1-plugins-must-be-compatible-with-the-gnu-general-public-license), not just a GPL-based license, it would be a challenging problem to maintain a licenses compatibility matrix that works in this more general case.

**Comment by ryanwelcher** [comment]
What needs to be done here from a docs point of view?
